cookbook/views/api.py

```diff
@@ -1455,13 +1455,21 @@ def post(self, request, *args, **kwargs):
 
             url = serializer.validated_data.get('url', None)
             data = unquote(serializer.validated_data.get('data', None))
+
+            duplicate = False
+            if url:
+                # Check for existing recipes with provided url
+                existing_recipe = Recipe.objects.filter(source_url=url).first()
```

Comment:
Could use `exists` instead, to avoid fetching the (unused) object.

Reply:
is changed/reworked for the new version anyway but yes.